### DISCUSSION TITLE
Aeris forecast data problem

### DISCUSSION THREAD

**Discussion by dciarniello** [poblabs/weewx-belchertown#814] [Q&A]
A week or so ago I started getting errors parsing Aeris forecast data.  The log shows

ERROR weewx.reportengine: Caught unrecoverable exception in generator 'weewx.cheetahgenerator.CheetahGenerator'
ERROR weewx.reportengine:         ****  'NoneType' object has no attribute 'split'
ERROR weewx.reportengine:         ****  Traceback (most recent call last):
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 197, in run
ERROR weewx.reportengine:         ****      obj.start()
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 385, in start
ERROR weewx.reportengine:         ****      self.run()
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 169, in run
ERROR weewx.reportengine:         ****      ngen = self.generate(gen_dict[section_name], section_name, self.gen_ts)
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 229, in generate
ERROR weewx.reportengine:         ****      ngen += self.generate(section[subsection], subsection, gen_ts)
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 229, in generate
ERROR weewx.reportengine:         ****      ngen += self.generate(section[subsection], subsection, gen_ts)
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 325, in generate
ERROR weewx.reportengine:         ****      _filename))
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 415, in _getSearchList
ERROR weewx.reportengine:         ****      search_list += obj.get_extension_list(timespan, db_lookup)
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/user/belchertown.py", line 1471, in get_extension_list
ERROR weewx.reportengine:         ****      data["current"][0]["response"]["ob"]["weatherPrimaryCoded"]
ERROR weewx.reportengine:         ****    File "/usr/share/weewx/user/belchertown.py", line 1066, in aeris_coded_weather
ERROR weewx.reportengine:         ****      coverage_code = data.split(":")[0]
ERROR weewx.reportengine:         ****  AttributeError: 'NoneType' object has no attribute 'split'
ERROR weewx.reportengine:         ****  Generator terminated

The error makes sense since weatherPrimaryCoded is, as can be seen from the attached file, indeed null. 

The question is, why is this suddenly happening and is there any fix for it?  

[forecast.json.gz](https://github.com/poblabs/weewx-belchertown/files/9367858/forecast.json.gz)

**Comment by norecords** [comment]
Did you try with another METAR?

**Reply by dciarniello** [reply]
I'm not sure what you mean by another METAR.  If you mean a provider other than Aeris, that wouldn't address the issue.  Aeris was working just fine for quite some time but suddenly broke without any changes on my side.  That suggests a change in what Aeris produces.

A number of questions come to mind:
1. Is it a bug in Aeris?
2. Is it a bug in weewx?
3. Is it a bug in Belchertown?
4. Is it a change in the Aeris API that needs to be handled by either weewx or Belchertown?
5. If the last, has it been handled and, if so, what do I need to update on my system?

**Reply by norecords** [reply]
Sorry for late reply,  what latitude and longitude value have you in your weewx.conf file and do you try with METAR = 1 or 0 to your skin.con? When I try your location with the Belchertown.py Aeris query I got no issue..

**Reply by dciarniello** [reply]
It seems that it was a temporary issue as it started working again a couple of weeks ago or so.
Given that I haven't changed anything, I'm guessing that it was a problem with Aeris potentially related, as you infer, to my lat/lon (~ 49N/123W).
